packages/native-stack/src/types.tsx

```diff
@@ -609,6 +631,16 @@ export type NativeStackNavigationOptions = {
    * Only supported on iOS and Android.
    */
   freezeOnBlur?: boolean;
+  /**
+   * Footer component that can be used alongside form sheet stack presentation style.
+   *
+   * This option is provided, because due to implementation details it might be problematic
+   * to implement such layout with JS-only code.
+   *
+   * Please note that this prop is marked as unstable and might be subject of breaking changes,
+   * including removal.
+   */
+  unstable_footerComponent?: React.ReactNode;
```

Comment:
I wouldn't call it `footerComponent` coz you're not accepting a React Component (`MyComponent` - component, `<MyComponent />` - element).

Though also in React Navigation, generally we accept render callbacks:

```js
unstable_footer?: () => React.ReactNode
```

This way we can pass additional information if we need in future. So maybe worth aligning it like that.